# 16s
**NCBI SRA #PRJEB27564**

    sratoolkit.3.1.1-centos_linux64/bin/fastq-dump --split-3  ERR3040013
    sratoolkit.3.1.1-centos_linux64/bin/fastq-dump --split-3  ERR2730388

# CO1

**NCBI SRA #PRJNA545426**

    sratoolkit.3.1.1-centos_linux64/bin/fastq-dump --split-3 SRR9185139
    sratoolkit.3.1.1-centos_linux64/bin/fastq-dump --split-3 SRR9185140
    sratoolkit.3.1.1-centos_linux64/bin/fastq-dump --split-3 SRR9185142

[Hajibabaei M, Porter T M, Wright M, et al. COI metabarcoding primer choice affects richness and recovery of indicator taxa in freshwater systems[J]. PLoS One, 2019, 14(9): e0220953.](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0220953)

# 12s
**NCBI SRA #PRJDB3526**

    sratoolkit.3.1.1-centos_linux64/bin/fastq-dump --split-3  DRR030411
    sratoolkit.3.1.1-centos_linux64/bin/fastq-dump --split-3  DRR030422

# 18s

**NCBI SRA #PRJNA592034**

    #v9
    sratoolkit.3.1.1-centos_linux64/bin/fastq-dump --split-3 SRR12793466
    sratoolkit.3.1.1-centos_linux64/bin/fastq-dump --split-3 SRR12793467
    #v4
    sratoolkit.3.1.1-centos_linux64/bin/fastq-dump --split-3 SRR12793385
    sratoolkit.3.1.1-centos_linux64/bin/fastq-dump --split-3 SRR12793390

# ITS

**NCBI SRA #PRJNA1055419**

    #primer:ITS4_ILL and gITS7_ILL   

    sratoolkit.3.1.1-centos_linux64/bin/fastq-dump --split-3 SRR30683883
    sratoolkit.3.1.1-centos_linux64/bin/fastq-dump --split-3 SRR30683886

[Rzehak T, Praeg N, Galla G, et al. Comparison of commonly used software pipelines for analyzing fungal metabarcoding data[J]. BMC genomics, 2024, 25(1): 1085.](https://link.springer.com/article/10.1186/s12864-024-11001-x)