# 16s
**NCBI SRA #PRJEB27564**

    sratoolkit.3.1.1-centos_linux64/bin/fastq-dump --split-3  ERR3040013
    sratoolkit.3.1.1-centos_linux64/bin/fastq-dump --split-3  ERR2730388
[Aho V T E, Pereira P A B, Voutilainen S, et al. Gut microbiota in Parkinson's disease: temporal stability and relations to disease progression[J]. EBioMedicine, 2019, 44: 691-707.](https://pmc.ncbi.nlm.nih.gov/articles/PMC6606744/)

# CO1

**NCBI SRA #PRJNA545426**

    sratoolkit.3.1.1-centos_linux64/bin/fastq-dump --split-3 SRR9185139
    sratoolkit.3.1.1-centos_linux64/bin/fastq-dump --split-3 SRR9185140
    sratoolkit.3.1.1-centos_linux64/bin/fastq-dump --split-3 SRR9185142
[Hajibabaei M, Porter T M, Wright M, et al. COI metabarcoding primer choice affects richness and recovery of indicator taxa in freshwater systems[J]. PLoS One, 2019, 14(9): e0220953.](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0220953)

# 12s
**NCBI SRA #PRJDB3526**

    sratoolkit.3.1.1-centos_linux64/bin/fastq-dump --split-3  DRR030411
    sratoolkit.3.1.1-centos_linux64/bin/fastq-dump --split-3  DRR030422
[Miya M, Sato Y, Fukunaga T, et al. MiFish, a set of universal PCR primers for metabarcoding environmental DNA from fishes: detection of more than 230 subtropical marine species[J]. Royal Society open science, 2015, 2(7): 150088.](https://royalsocietypublishing.org/doi/full/10.1098/rsos.150088)

# 18s

**NCBI SRA #PRJNA592034**

    #v9
    sratoolkit.3.1.1-centos_linux64/bin/fastq-dump --split-3 SRR12793466
    sratoolkit.3.1.1-centos_linux64/bin/fastq-dump --split-3 SRR12793467
    #v4
    sratoolkit.3.1.1-centos_linux64/bin/fastq-dump --split-3 SRR12793385
    sratoolkit.3.1.1-centos_linux64/bin/fastq-dump --split-3 SRR12793390
[Choi J, Park J S. Comparative analyses of the V4 and V9 regions of 18S rDNA for the extant eukaryotic community using the Illumina platform[J]. Scientific Reports, 2020, 10(1): 6519.](https://www.nature.com/articles/s41598-020-63561-z)

# ITS

**NCBI SRA #PRJNA1055419**

    #primer:ITS4_ILL and gITS7_ILL   

    sratoolkit.3.1.1-centos_linux64/bin/fastq-dump --split-3 SRR30683883
    sratoolkit.3.1.1-centos_linux64/bin/fastq-dump --split-3 SRR30683886
[Rzehak T, Praeg N, Galla G, et al. Comparison of commonly used software pipelines for analyzing fungal metabarcoding data[J]. BMC genomics, 2024, 25(1): 1085.](https://link.springer.com/article/10.1186/s12864-024-11001-x)


# rbcL

**NCBI SRA #PRJNA650007**

    #primer:rbcL2-F and rbcLa-R primers
    
    sratoolkit.3.1.1-centos_linux64/bin/fastq-dump --split-3 SRR12374561
    sratoolkit.3.1.1-centos_linux64/bin/fastq-dump --split-3 SRR12374563

[Urumarudappa S K J, Tungphatthong C, Prombutara P, et al. DNA metabarcoding to unravel plant species composition in selected herbal medicines on the National List of Essential Medicines (NLEM) of Thailand[J]. Scientific Reports, 2020, 10(1): 18259.](https://www.nature.com/articles/s41598-020-75305-0)
