# Observing_Life_in_the Sea_Using_Environmental_DNA

![Observing Life in the Sea Using Environmental DNA](./Observing_Life_in_the_Sea_Using_Environmental_DNA.jpg)

## ITS region

![ITS region](./ITS/ITS-region.png)

![ITS](./ITS/ITS1-F_ITS86R_ITS86F_ITS4.jpeg)

[Vancov T, Keen B. Amplification of soil fungal community DNA using the ITS86F and ITS4 primers[J]. FEMS microbiology letters, 2009, 296(1): 91-96.](https://academic.oup.com/femsle/article/296/1/91/485778?login=true)

[Bellemain E, Carlsen T, Brochmann C, et al. ITS as an environmental DNA barcode for fungi: an in silico approach reveals potential PCR biases[J]. BMC microbiology, 2010, 10: 1-9.](https://link.springer.com/article/10.1186/1471-2180-10-189)

[Op De Beeck M, Lievens B, Busschaert P, et al. Comparison and validation of some ITS primer pairs useful for fungal metabarcoding studies[J]. PloS one, 2014, 9(6): e97629.](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0097629)

## 16s 

![16s](./16s/16S_ribosomal_RNA_gene.png)

[Fadeev E, Cardozo-Mino M G, Rapp J Z, et al. Comparison of two 16S rRNA primers (V3–V4 and V4–V5) for studies of arctic microbial communities[J]. Frontiers in microbiology, 2021, 12: 637526.](https://www.frontiersin.org/journals/microbiology/articles/10.3389/fmicb.2021.637526/full)

[Babis W, Jastrzebski J P, Ciesielski S. Fine-Tuning of DADA2 Parameters for Multiregional Metabarcoding Analysis of 16S rRNA Genes from Activated Sludge and Comparison of Taxonomy Classification Power and Taxonomy Databases[J]. International Journal of Molecular Sciences, 2024, 25(6): 3508.](https://www.mdpi.com/1422-0067/25/6/3508)

[Na H S, Song Y, Yu Y, et al. Comparative analysis of primers used for 16S rRNA gene sequencing in oral microbiome studies[J]. Methods and Protocols, 2023, 6(4): 71.](https://www.mdpi.com/2409-9279/6/4/71)

## 18s rRNA

![18s](./18s/18s.png)

V4 (expected amplicon size, 270 bp–387bp) and V9 (expected amplicon size, 96 bp–134bp)

[Zheng X, He Z, Wang C, et al. Evaluation of different primers of the 18S rRNA gene to profile amoeba communities in environmental samples[J]. Water Biology and Security, 2022, 1(3): 100057.](https://www.sciencedirect.com/science/article/pii/S2772735122000774)

## 12s

![12s_v5](./12s/12s_RiazV5_MiFish_Tele02.png)

![12s_teleo](./12s/Schematic-representation-of-a-the-mitochondrial-mitogenome-with-the-12S-mitochondrial.png)

[Oliveira Carvalho C, Pazirgiannidi M, Ravelomanana T, et al. Multi-method survey rediscovers critically endangered species and strengthens Madagascar's freshwater fish conservation[J]. Scientific Reports, 2024, 14(1): 20427.](https://www.nature.com/articles/s41598-024-71398-z)

[Miya M, Sato Y, Fukunaga T, et al. MiFish, a set of universal PCR primers for metabarcoding environmental DNA from fishes: detection of more than 230 subtropical marine species[J]. Royal Society open science, 2015, 2(7): 150088.](https://royalsocietypublishing.org/doi/full/10.1098/rsos.150088)

## CO1

**mlCOIintF_HCO2198**

5′-GGWACWGGWTGAACWGTWTAYCCYCC-3′

5′-TAAACTTCAGGGTGACCAAARAAYCA-3′

![CO1](./CO1/CO1.png)

[Hajibabaei M, Porter T M, Wright M, et al. COI metabarcoding primer choice affects richness and recovery of indicator taxa in freshwater systems[J]. PLoS One, 2019, 14(9): e0220953.](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0220953)

## [Earth Microbiome Project](https://earthmicrobiome.org/) 

- [16S Illumina Amplicon Protocol:v4-(515F–806R) V4-V5 (515F-926R)](https://earthmicrobiome.org/protocols-and-standards/16s/)
- [18S Illumina Amplicon Protocol:v9-(Euk1391f-EukBr)](https://earthmicrobiome.org/protocols-and-standards/18s/)
- [ITS Illumina Amplicon Protocol:(ITS1f-ITS2)](https://earthmicrobiome.org/protocols-and-standards/its/)

| Name	                      |Forward|	Reverse	|Amplicon_max_Length(bp)|	Notes|
|----------------------------|---|---|---|---|
| 12s_rRNA_MiFish-U_MiFish-L |	GTCGGTAAAACTCGTGCCAGC|	CATAGTGGGGTATCTAATCCCAGTTTG	|185|	https://github.com/billzt/MiFish,Miya M, Sato Y, Fukunaga T, et al. MiFish, a set of universal PCR primers for metabarcoding environmental DNA from fishes: detection of more than 230 subtropical marine species[J]. Royal Society open science, 2015, 2(7): 150088.|
 |12s_rRNA_V5_Riaz           |	ACTGGGATTAGATACCCC|	TAGAACAGGCTCCTCTAG|	117|	Oliveira Carvalho C, Pazirgiannidi M, Ravelomanana T, et al. Multi-method survey rediscovers critically endangered species and strengthens Madagascar's freshwater fish conservation[J]. Scientific Reports, 2024, 14(1): 20427.](https://www.nature.com/articles/s41598-024-71398-z.|
|12s_rRNA_Tele02|	AAACTCGTGCCAGCCACC|	GGGTATCTAATCCCAGTTTG|	209|	Oliveira Carvalho C, Pazirgiannidi M, Ravelomanana T, et al. Multi-method survey rediscovers critically endangered species and strengthens Madagascar's freshwater fish conservation[J]. Scientific Reports, 2024, 14(1): 20427.](https://www.nature.com/articles/s41598-024-71398-z)|
|Fish_16s_rRNA_F/D_16S2R|	GACCCTATGGAGCTTTAGAC|	CGCTGTTATCCCTADRGTAATC|	200|	Adams C I M, Jeunen G J, Cross H, et al. Environmental DNA metabarcoding describes biodiversity across marine gradients[J]. ICES Journal of Marine Science, 2023, 80(4): 953-971.|
|Crustacean_Crust16S_F_Crust16S_R|	GGGACGATAAGACCCTATA	|ATTACGCTGTTATCCCTAAAG|	200	|Adams C I M, Jeunen G J, Cross H, et al. Environmental DNA metabarcoding describes biodiversity across marine gradients[J]. ICES Journal of Marine Science, 2023, 80(4): 953-971.|
|16s_rRNA_V1-V2_27F-338R|	AGAGTTTGATYMTGGCTCAG	|TGCTGCCTCCCGTAGRAGT	|311|	Babis W, Jastrzebski J P, Ciesielski S. Fine-Tuning of DADA2 Parameters for Multiregional Metabarcoding Analysis of 16S rRNA Genes from Activated Sludge and Comparison of Taxonomy Classification Power and Taxonomy Databases[J]. International Journal of Molecular Sciences, 2024, 25(6): 3508.|
|16s_rRNA_V1-V3_27F-534R|	AGAGTTTGATYMTGGCTCAG	|ATTACCGCGGCTGCTGG	|500|	Babis W, Jastrzebski J P, Ciesielski S. Fine-Tuning of DADA2 Parameters for Multiregional Metabarcoding Analysis of 16S rRNA Genes from Activated Sludge and Comparison of Taxonomy Classification Power and Taxonomy Databases[J]. International Journal of Molecular Sciences, 2024, 25(6): 3508.|
|16s_rRNA_V3-v4_341F-785R|	CCTACGGGNGGCWGCAG	|GACTACHVGGGTATCTAATCC	|470|	Babis W, Jastrzebski J P, Ciesielski S. Fine-Tuning of DADA2 Parameters for Multiregional Metabarcoding Analysis of 16S rRNA Genes from Activated Sludge and Comparison of Taxonomy Classification Power and Taxonomy Databases[J]. International Journal of Molecular Sciences, 2024, 25(6): 3508.|
|16s_rRNA_V3-v4_341F-805R|	CCTACGGGNBGCASCAG	|GACTACNVGGGTATCTAATCC	|470|	Babis W, Jastrzebski J P, Ciesielski S. Fine-Tuning of DADA2 Parameters for Multiregional Metabarcoding Analysis of 16S rRNA Genes from Activated Sludge and Comparison of Taxonomy Classification Power and Taxonomy Databases[J]. International Journal of Molecular Sciences, 2024, 25(6): 3508.|
|16s_rRNA_V4-V5_515F–944R|	GTGYCAGCMGCCGCGGTAA	|CCGYCAATTYMTTTRAGTTT	|430|	Fadeev E, Cardozo-Mino M G, Rapp J Z, et al. Comparison of two 16S rRNA primers (V3–V4 and V4–V5) for studies of arctic microbial communities[J]. Frontiers in microbiology, 2021, 12: 637526.|
|16s_rRNA_V4_515F–806R|	GTGYCAGCMGCCGCGGTAA	|GGACTACNVGGGTWTCTAAT	|253|	https://earthmicrobiome.org/protocols-and-standards/16s/,Zhao J, Rodriguez J, Martens-Habbena W. Fine-scale evaluation of two standard 16S rRNA gene amplicon primer pairs for analysis of total prokaryotes and archaeal nitrifiers in differently managed soils[J]. Frontiers in Microbiology, 2023, 14: 1140487.|
|16s_rRNA_V4-V5_515F–926R|	GTGYCAGCMGCCGCGGTAA	|CCGYCAATTYMTTTRAGTTT	|374|	https://earthmicrobiome.org/protocols-and-standards/16s/,Zhao J, Rodriguez J, Martens-Habbena W. Fine-scale evaluation of two standard 16S rRNA gene amplicon primer pairs for analysis of total prokaryotes and archaeal nitrifiers in differently managed soils[J]. Frontiers in Microbiology, 2023, 14: 1140487.|
|ITS1_ITS1F-ITS2|	CTTGGTCATTTAGAGGAAGTAA	|GCTGCGTTCTTCATCGATGC	|300|	https://earthmicrobiome.org/protocols-and-standards/its/,Op De Beeck M, Lievens B, Busschaert P, et al. Comparison and validation of some ITS primer pairs useful for fungal metabarcoding studies[J]. PloS one, 2014, 9(6): e97629.|
|ITS1_ITS1F-ITS86R|	CTTGGTCATTTAGAGGAAGTAA	|TTCAAAGATTCGATGATTCAG	|300|	https://earthmicrobiome.org/protocols-and-standards/its/,Op De Beeck M, Lievens B, Busschaert P, et al. Comparison and validation of some ITS primer pairs useful for fungal metabarcoding studies[J]. PloS one, 2014, 9(6): e97629.|
|ITS2_ITS3-ITS4|	GCATCGATGAAGAACGCAGC	|TCCTCCGCTTATTGATATGC	|450|	Vancov T, Keen B. Amplification of soil fungal community DNA using the ITS86F and ITS4 primers[J]. FEMS microbiology letters, 2009, 296(1): 91-96.|
|ITS2_ITS7-ITS4|	GTGARTCATCGARTCTTTG	|TCCTCCGCTTATTGATATGC	|400|	Op De Beeck M, Lievens B, Busschaert P, et al. Comparison and validation of some ITS primer pairs useful for fungal metabarcoding studies[J]. PloS one, 2014, 9(6): e97629.|
|ITS2_ITS86F-ITS4|	GTGAATCATCGAATCTTTGAA	|TCCTCCGCTTATTGATATGC	|450|	Vancov T, Keen B. Amplification of soil fungal community DNA using the ITS86F and ITS4 primers[J]. FEMS microbiology letters, 2009, 296(1): 91-96.|
|18s_rRNA_V4_TAReuk454FWD1-TAReukREV3|	GTGAATCATCGAATCTTTGAA	|ACTTTCGTTCTTGATYRATGA|	420	|Chun S J. Microbiome dataset of eukaryotic and fungal communities in the bulk soil and root of wild Brassica napus in South Korea[J]. Data in Brief, 2022, 43: 108457.|
|18s_rRNA_V9_1380F-1510R|	CCCTGCCHTTTGTACACAC	|CCTTCYGCAGGTTCACCTAC|	130|	Zheng X, He Z, Wang C, et al. Evaluation of different primers of the 18S rRNA gene to profile amoeba communities in environmental samples[J]. Water Biology and Security, 2022, 1(3): 100057.|
|18s_rRNA_V9_1391F-EukBr|	GTACACACCGCCCGTC|	TGATCCTTCTGCAGGTTCACCTAC|	130|	https://earthmicrobiome.org/protocols-and-standards/18s/.|
|CO1_mlCOIintF-dgHCO2198|	GGWACWGGWTGAACWGTWTAYCCYCC|	TAAACTTCAGGGTGACCAAARAAYCA|	313|	https://naturalhistory.si.edu/sites/default/files/media/file/arms-12coimetabarcoding.pdf|
